carbon-mb/components/andes-rest-service/src/main/java/org/wso2/carbon/andes/service/internal/AndesRESTService.java

```diff
@@ -1333,13 +1308,18 @@ public Response deleteMessages(
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = "Successfully received cluster information.",
                          response = ClusterInformation.class),
+            @ApiResponse(code = 404, message = "Clustering information not found.", response = ErrorResponse.class),
             @ApiResponse(code = 500, message = "Server Error.", response = ErrorResponse.class)})
-    public Response getClusterInformation() {
+    public Response getClusterInformation() throws InternalServerException {
         try {
             ClusterInformation clusterInformation = brokerManagerService.getClusterInformation();
-            return Response.status(Response.Status.OK).entity(clusterInformation).build();
+            if (null != clusterInformation) {
+                return Response.status(Response.Status.OK).entity(clusterInformation).build();
+            } else {
+                throw new BrokerManagerException("Unable to find cluster information.");
+            }
         } catch (BrokerManagerException e) {
-            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(e).build();
+            throw new InternalServerException(e);
```

Comment:
Wanted to pass the original exception message to InternalServerException class, thats why I passed the exception.